FireboltNETSDK/FireboltClient.cs

```diff
@@ -226,6 +238,44 @@ protected async Task<T> SendAsync<T>(HttpMethod method, string uri, HttpContent?
 
         try
         {
+            string updateParametersHeader = "Firebolt-Update-Parameters";
+            if (response.Headers.Contains(updateParametersHeader))
+            {
+                string? newDatabase = null;
+                string? newEngine = null;
+                bool reconnect = false;
+                foreach (string[] kv in response.Headers.GetValues(updateParametersHeader).Select(p => p.Split('=', 2, StringSplitOptions.TrimEntries)))
+                {
+                    if (kv.Length == 1 || "".Equals(kv[1]))
+                    {
+                        connectionParameters.Remove(kv[0]);
+                    }
+                    else
+                    {
+                        connectionParameters[kv[0]] = kv[1];
+                        switch (kv[0])
+                        {
+                            case "database":
+                                newDatabase = kv[1];
+                                reconnect = true;
+                                break;
```

Comment:
This logic can probably be refactored into a separate method. `SendAsync` already feels quite bloated.